packages/@lwc/perf-benchmarks/rollup.config.mjs

```diff
@@ -20,22 +19,20 @@ const componentModules = glob
     .map((filename) => path.relative(path.join(__dirname, '../..'), filename).replace(/^\.\//, ''));
 
 function createConfig(benchmarkFile) {
-    const isServer = benchmarkFile.includes('/engine-server/');
-    const lwcImportModule = isServer ? '@lwc/engine-server' : '@lwc/engine-dom';
-
+    // Pseudo-Best benchmark framework implementing globals `before`, `after`, `benchmark`, `run`
+    const benchmarkFramework = path.resolve(__dirname, './src/utils/benchmark-framework.js');
     return {
         input: benchmarkFile,
         plugins: [
-            !isServer && syntheticShadow(),
-            // Replace `import { foo } from 'lwc'` with '@lwc/engine-server' or '@lwc/engine-dom'
-            replace({
-                preventAssignment: true,
-                delimiters: ['', ''],
-                values: {
-                    'from "lwc"': `from "${lwcImportModule}"`,
-                    "from 'lwc'": `from "${lwcImportModule}"`,
-                },
-            }),
```

Comment:
Rather than having a complicated system for replacing `"lwc"` with `"@lwc/engine-dom"` or `"@lwc/engine-server"`, we can just be explicit in the benchmark files. They are either dom or server – none of them support both. And the components from `@lwc/perf-benchmarks` are already precompiled with Rollup, so they don't import from `"lwc"`.